tests/types.ts

```diff
@@ -170,3 +170,36 @@ let Rn: Reacts<number>
   const s: string = r[0]
   const l: boolean = a.map(s => s.length).every(n => n < 2).valueOf()
 }
+// properties
+{
+  let o = {
+    str: 'string',
+    num: 1,
+    vb: new Variable(true),
+    inner: new Variable({
+      x: 2,
+      y: new Variable({
+        a: false
+      })
+    })
+  }
+  let vo = new Variable(o)
+  const s: string = vo.get('str')
+  const num: number = vo.get('num')
+  const b: boolean = vo.get('vb')
+  const inner = vo.get('inner')
+  const x: number = inner.x
+  //const y = inner.y
+  //const yab: boolean = y.a
```

Comment:
i think this maybe actually produces the entirely flattened object, but I think it's a sufficient improvement to get the single level of type inference